include/fixed_containers/fixed_list.hpp

```diff
@@ -0,0 +1,963 @@
+                        const T& value,
+                        const std_transition::source_location& loc =
+                            std_transition::source_location::current()) noexcept
+      : Base(count, value, loc)
+    {
+    }
+    constexpr explicit FixedList(std::size_t count,
+                                 const std_transition::source_location& loc =
+                                     std_transition::source_location::current()) noexcept
+      : Base(count, loc)
+    {
+    }
+    template <InputIterator InputIt>
+    constexpr FixedList(InputIt first,
+                        InputIt last,
+                        const std_transition::source_location& loc =
+                            std_transition::source_location::current()) noexcept
+      : Base(first, last, loc)
+    {
+    }
+    constexpr FixedList(std::initializer_list<T> list,
+                        const std_transition::source_location& loc =
+                            std_transition::source_location::current()) noexcept
+      : Base(list, loc)
+    {
+    }
+
+    constexpr FixedList(const FixedList& other)
+        requires TriviallyCopyConstructible<T>
+    = default;
+    constexpr FixedList(FixedList&& other) noexcept
+        requires TriviallyMoveConstructible<T>
+    = default;
+    constexpr FixedList& operator=(const FixedList& other)
+        requires TriviallyCopyAssignable<T>
+    = default;
+    constexpr FixedList& operator=(FixedList&& other) noexcept
+        requires TriviallyMoveAssignable<T>
+    = default;
+
+    constexpr FixedList(const FixedList& other)
+      : FixedList(other.begin(), other.end())
+    {
+    }
+    constexpr FixedList(FixedList&& other) noexcept
+      : FixedList()
+    {
+        for (T& entry : other)
+        {
+            this->push_back_internal(std::move(entry));
+        }
+
+        // Clear the moved-out-of-list. This is consistent with both std::list
+        // as well as the trivial move constructor of this class.
+        other.clear();
+    }
+    constexpr FixedList& operator=(const FixedList& other)
+    {
+        if (this == &other)
+        {
+            return *this;
+        }
+
+        this->assign(other.begin(), other.end());
+        return *this;
+    }
+    constexpr FixedList& operator=(FixedList&& other) noexcept
+    {
+        if (this == &other)
+        {
+            return *this;
+        }
+
+        this->clear();
+        for (T& entry : other)
+        {
+            this->push_back_internal(std::move(entry));
+        }
+        // The trivial assignment operator does not `other.clear()`, so don't do it here either for
+        // consistency across FixedLists. std::list<T> does clear it, so behavior is different.
+        // Both choices are fine, because the state of a moved object is intentionally unspecified
+        // as per the standard and use-after-move is undefined behavior.
+        return *this;
+    }
+
+    constexpr ~FixedList() noexcept { this->clear(); }
```

Comment:
I think the move, copy, and destruction strangeness should be handled in the `FixedDoublyLinkedList` strangeness, not this "client" class. That way, `FixedDoublyLinkedList` is "feature complete" for use in other containers (e.g. `FixedUnorderedMap`)

Reply:
Will put memory management semantics to this.